azurelinuxagent/pa/deprovision/default.py

```diff
@@ -111,6 +111,14 @@ def del_resolv(self, warnings, actions):
 
     def del_dhcp_lease(self, warnings, actions):
         warnings.append("WARNING! Cached DHCP leases will be deleted.")
+
+        # For Arch based systems. Kill DHCPCD service. Needed to clear cached leases.
+        shellutil.run("systemctl stop dhcpcd", chk_err=False)
+
+        # For Arch based systems. Files mounted by DHCPCD can't be deleted whilst mounted, not even by root.
+        dirs_to_umount = ["/var/lib/dhcpcd/run/systemd/journal", "/var/lib/dhcpcd/run/udev", "/var/lib/dhcpcd/sys", "/var/lib/dhcpcd/dev", "/var/lib/dhcpcd/proc"]
+        actions.append(DeprovisionAction(fileutil.umount, dirs_to_umount))
```

Comment:
If this is Arch-specific logic, then it should go into the `ArchDeprovisionHandler`.

Reply:
So perhaps I should rephrase my comment here. While it's true this is for Arch systems, it's also needed for any *Nix distro that's running the latest version of `dhcpcd`. A quick check on wiki shows that a large number  of distros use `systemd/dhcpcd` by default or as an optional configuration:
![Screenshot_2020-06-26_17-18-33](https://user-images.githubusercontent.com/31227781/85909949-2c72f280-b7d1-11ea-96b5-a6d777fc6843.png)


The systems that can be configured with `systemd/dhcpcd` include: Arch Linux, CentOS, CoreOS, Fedora, Mageia, Manjaro Linux, openSUSE, SUSE Linux Enterprise Server, Red Hat Enterprise Linux, Solus, Linux Mint, Ubuntu and Debian.


Among those, many use `systemd/dhcpcd` by default and with none/few alternatives.


I believe this is the correct place to run this, but I can update my comment to make it clear this is a DHCPCD specific issue and not an Arch one.